sdk/eventhub/Azure.Messaging.EventHubs/src/Primitives/EventProcessor.cs

```diff
@@ -1717,17 +1745,36 @@ private async Task RunProcessingAsync(CancellationToken cancellationToken)
             }
             catch (Exception ex) when (ex.IsFatalException())
             {
+                // It is not safe to allocate or log here, as this class of errors includes those such as
+                // OutOfMemoryException and StackOverflowException.  It is assumed the process is crashing.
+
                 throw;
             }
             catch (Exception ex)
```

Comment:
With the new inner structure, the only case that should get us here would be if the connection factory throws.  The same factory is used by validation, so the processor will refuse to start unless the factory failure is intermittent.   _(Note: this is not establishing/opening the connection, just instantiating the type)_